velox/functions/prestosql/types/JsonType.cpp

```diff
@@ -53,6 +53,8 @@ void generateJsonTyped(
     } else if constexpr (
         std::is_same_v<T, Date> || std::is_same_v<T, Timestamp>) {
       result.append(std::to_string(value));
+    } else if constexpr (std::is_same_v<T, ShortDecimal>) {
+      VELOX_UNSUPPORTED();
```

Comment:
Probably write a comment explaining that this is unsupported since ShortDecimal requires precision and scale which are part of the Type.